test/infrastructure/docker/docker/types/node.go

```diff
@@ -83,6 +83,9 @@ func (n *Node) IP(ctx context.Context) (ipv4 string, ipv6 string, err error) {
 	if len(ips) != 2 {
 		return "", "", errors.Errorf("container addresses should have 2 values, got %d values", len(ips))
 	}
+	if ips[0] == "" && ips[1] == "" {
```

Comment:
This change seems unrelated to the LB container change to me. Am I missing something obvious?

Reply:
In case a change here was intended. Do we want to fail if:
* both are empty
* one of them are empty

Reply:
@MarcelMue This change is related to the LB container.
During DockerCluster reconciliation:
1. In the [reconcileNormal](https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/controllers/dockercluster_controller.go#L144) method we try to look up the IP of the LB of the previously provisioned LB by calling the [LoadBalancer.IP](https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/docker/loadbalancer.go#L135)
2. [LoadBalancer.IP](https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/docker/loadbalancer.go#L135) in turn calls the [Node.IP](https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/docker/types/node.go#L69) function by calling [s.container.IP()](https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/docker/loadbalancer.go#L136)

Reply:
@sbueringer the IPs returned here are IPv4 (ips[0]) and IPv6(ips[1] addresses. Till we decide that only one of those is what we will support, returning an error if both are empty seems reasonable. WDYT?

Reply:
On thinking about this further, returning this error from the `Node.IP()` may not be the correct thing. IMO, this check should be performed in the `LoadBalancer.IP` function.

Reply:
@ashish-amarnath Sounds okay to me. I don't have the necessary context to know if we expect both of them to be set or just one of them.